### PR TITLE
Jakarta9 source updates

### PR DESCRIPTION
**Pull request by sberyozkin**

### CODE REVIEW

**Comment by sberyozkin** [comment]
@Emily-Jiang @radcortez never mind, I haven't rebased the branch

**Comment by sberyozkin** [comment]
@Emily-Jiang @radcortez Something has changed in the `2.1` parent so now `// @formatter off` in `api/src/main/java/org/eclipse/microprofile/jwt/Claims.java` is ignored and it is misformatted. Do you know what it can be ?

**Comment by Emily-Jiang** [comment]
[Here](https://github.com/eclipse/microprofile-parent/commits/master) are the changes in microprofile-parent

**Comment by sberyozkin** [comment]
Hi @Emily-Jiang - if you have some time, can you please checkout this PR's branch - and independently verify `Claims.java` is affected ?
Please also note I had to revert some TCK source file changes:
https://github.com/sberyozkin/microprofile-jwt-auth/commit/7eec086a1d84889dfc2a4adab3052d48ad214eb5
(EJB, crypto, servlet classes and possibly a few others have no `jakarta.*` equivalents) - not sure if it is parent problem or something else
I think the package.info update (see the last commit) is fine, but check please

thanks

**Comment by sberyozkin** [comment]
Note in https://github.com/sberyozkin/microprofile-jwt-auth/commit/7eec086a1d84889dfc2a4adab3052d48ad214eb5 the package import order is changed again after building the TCK, so the actual fixes `from jakarta.* back to javax.*` are not very easy to see

File: spec/src/main/asciidoc/interoperability.asciidoc
Comment (sberyozkin) [inline spec/src/main/asciidoc/interoperability.asciidoc:682]: @Emily-Jiang If some of `jakarta.* -> javax.*` revert updates in TCK are correct then I'd need to update this doc as well to retain `javax.ejb.SessionContext` for ex. Commenting here so that I don't forget

**Review by Emily-Jiang — APPROVED** [review]
LGTM!

**Comment by Emily-Jiang** [comment]
@sberyozkin please feel free to merge your PR.

**Comment by Emily-Jiang** [comment]
> @Emily-Jiang @radcortez Something has changed in the `2.1` parent so now `// @formatter off` in `api/src/main/java/org/eclipse/microprofile/jwt/Claims.java` is ignored and it is misformatted when the source is built. Do you know what it can be ?

I checked out your PR to try it out. I think `//@formatter off` is only a tag for telling IDEs to ignore the formatting. Since we are using maven formatter plugin, the annotation was completely ignored.

**Comment by Emily-Jiang** [comment]
You can merge in this change. I spotted more changes to be done to pull in the 3.0-RC4 from MP Config and also the dependency on EJB needs to be updated. In your pom.xml, you need to update the following 

        ```
<version.jakarta.ejb.api>3.2.6</version.jakarta.ejb.api>
<version.microprofile.config>2.0</version.microprofile.config>
```
=>

       ```
 <version.jakarta.ejb.api>4.0.0</version.jakarta.ejb.api>
 <version.microprofile.config>3.0-RC4</version.microprofile.config>
```
You can do this change under the issue #252

File: spec/src/main/asciidoc/interoperability.asciidoc
Comment (rdebusscher) [reply spec/src/main/asciidoc/interoperability.asciidoc:682]: I don't understand @sberyozkin . Do you mean that SessionContext will still be used from the javax namespace? It is also converted to jakarta within Jakarta 9;  https://jakarta.ee/specifications/platform/9/apidocs/jakarta/ejb/sessioncontext

**Review by rdebusscher — CHANGES_REQUESTED** [review]

**Comment by sberyozkin** [comment]
@rdebusscher I've no idea, the compilation was failing with the jakarta package - may be what Emily suggested can help

**Comment by sberyozkin** [comment]
@Emily-Jiang Thanks, but I don't want to release before the compilation succeeds, there is no much point in it :-)

> I checked out your PR to try it out. I think //@formatter off is only a tag for telling IDEs to ignore the formatting. Since we are using maven formatter plugin, the annotation was completely ignored.

It works fine on `master` and likewise it should work with the `2.0.0-RC1` so I'd not like to release before it can be sorted out

thanks

File: spec/src/main/asciidoc/interoperability.asciidoc
Comment (sberyozkin) [reply spec/src/main/asciidoc/interoperability.asciidoc:682]: @rdebusscher The compilation was failing, I'll try @Emily-Jiang's suggestion a bit later on

**Comment by sberyozkin** [comment]
@Emily-Jiang @rdebusscher It compiles with the EJB version update - Rudy if all is good otherwise then please dismiss your change request

**Comment by sberyozkin** [comment]
@Emily-Jiang yes, thanks for noticing those old versions. The only problem is the Claims misformatting but I guess fixing it can wait

**Comment by Emily-Jiang** [comment]
> @Emily-Jiang yes, thanks for noticing those old versions. The only problem is the Claims misformatting but I guess fixing it can wait

The formatter is pretty much personal taste. For me, the refomatted version looks nicer:). I think after merging this PR, we can do a 2.0-RC1. If you find a solution to fix the formatter, we do a RC2 release. In this way, it will not block the implementation to try out the jars.

**Comment by rdebusscher** [comment]
@sberyozkin The  version of JACC is not updated (and still is the javax variant) for the TCK.



we need to replace

```
            <dependency>
                <groupId>jakarta.security.jacc</groupId>
                <artifactId>jakarta.security.jacc-api</artifactId>
                <version>${version.jakarta.security.jacc-api}</version>
                <scope>provided</scope>
                <optional>true</optional>
            </dependency>
```
with

```
<dependency>
  <groupId>jakarta.authorization</groupId>
  <artifactId>jakarta.authorization-api</artifactId>
  <version>2.0.0</version>
  <scope>provided</scope>
  <optional>true</optional>
</dependency>
```

The javax JACC artifact also brings in the javax servlet one which means there are still references to javax.servlet.* in the TCK code.

**Comment by sberyozkin** [comment]
@rdebusscher let me try, thanks

**Comment by sberyozkin** [comment]
@rdebusscher It should be fine now...

**Comment by Emily-Jiang** [comment]
> Hi @Emily-Jiang I agree it is a personal taste but I should say IMHO having 20 or so enums `stringified` is not good, we don't do it all for the users only but for the developers too. If it were 3 enums only then it would not be a big deal, but otherwise it is not cool:
> 
> ```
> // @formatter:off
> public enum Claims {
>     // The base set of required claims that MUST have non-null values in the JsonWebToken
>     iss("Issuer", String.class), sub("Subject", String.class), exp("Expiration Time", Long.class), iat("Issued At Time",
>             Long.class), jti("JWT ID", String.class), upn("MP-JWT specific unique principal name",
>                     String.class), groups("MP-JWT specific groups permission grant",
>                             Set.class), raw_token("MP-JWT specific original bearer token", String.class),
> ```
> 
> And this is just a single fragment of `Claims.java`, many of them are now there.
> I don't think we can offer it to the developers.
> 
> But like I said, I'm fine with doing RC2 once the solution is found but it will need to be found before 2.0

I did not pay much attention to the `value` except to the style. You were right that the current style is lot more readable. I think one solution is to exclude this file from being reformatted. You can add the following to the api/pom.xml to skip that file:
```
<build> 
        <pluginManagement>
            <plugins>
                <plugin>
                    <groupId>net.revelc.code.formatter</groupId>
                    <artifactId>formatter-maven-plugin</artifactId>
                    <version>${version.plugin.formatter}</version>
                    <configuration>
                        <excludes>
                            <exclude>api/src/main/java/org/eclipse/microprofile/jwt/Claims.java</exclude> 
                        </excludes>
                    </configuration>
                </plugin>
            </plugins>
        </pluginManagement>
</build>


```
There might a better solution. Can you take a look to see whether there are no other side effects?

**Comment by sberyozkin** [comment]
Hi @Emily-Jiang I should've guess it myself :-), that should do, I'll try, thanks for trying to help

**Comment by rdebusscher** [comment]
Looks good, I only see some `javax` related code within the `JWT RBAC for MicroProfile Sandbox` but that is not a problem.

Thx

**Pull request merged**
